src/common/compress/zst/compress.c

```diff
@@ -0,0 +1,207 @@
+    IoFilter *filter;                                               // Filter interface
+
+    bool inputSame;                                                 // Is the same input required on the next process call?
+    size_t inputOffset;                                             // Current offset in input buffer
+    bool flushing;                                                  // Is input complete and flushing in progress?
+} ZstCompress;
+
+/***********************************************************************************************************************************
+Render as string for logging
+***********************************************************************************************************************************/
+static String *
+zstCompressToLog(const ZstCompress *this)
+{
+    return strNewFmt(
+        "{level: %d, inputSame: %s, inputOffset: %zu, flushing: %s}", this->level, cvtBoolToConstZ(this->inputSame),
+        this->inputOffset, cvtBoolToConstZ(this->flushing));
+}
+
+#define FUNCTION_LOG_ZST_COMPRESS_TYPE                                                                                             \
+    ZstCompress *
+#define FUNCTION_LOG_ZST_COMPRESS_FORMAT(value, buffer, bufferSize)                                                                \
+    FUNCTION_LOG_STRING_OBJECT_FORMAT(value, zstCompressToLog, buffer, bufferSize)
+
+/***********************************************************************************************************************************
+Free compression context
+***********************************************************************************************************************************/
+OBJECT_DEFINE_FREE_RESOURCE_BEGIN(ZST_COMPRESS, LOG, logLevelTrace)
+{
+    ZSTD_freeCStream(this->context);
+}
+OBJECT_DEFINE_FREE_RESOURCE_END(LOG);
+
+/***********************************************************************************************************************************
+Compress data
+***********************************************************************************************************************************/
+static void
+zstCompressProcess(THIS_VOID, const Buffer *uncompressed, Buffer *compressed)
+{
+    THIS(ZstCompress);
+
+    FUNCTION_LOG_BEGIN(logLevelTrace);
+        FUNCTION_LOG_PARAM(ZST_COMPRESS, this);
+        FUNCTION_LOG_PARAM(BUFFER, uncompressed);
+        FUNCTION_LOG_PARAM(BUFFER, compressed);
+    FUNCTION_LOG_END();
+
+    ASSERT(this != NULL);
+    ASSERT(!(this->flushing && !this->inputSame));
+    ASSERT(this->context != NULL);
+    ASSERT(compressed != NULL);
+    ASSERT(!this->flushing || uncompressed == NULL);
+
+    // Initialize output buffer
+    ZSTD_outBuffer out = {.dst = bufRemainsPtr(compressed), .size = bufRemains(compressed)};
+
+    // If input is NULL then start flushing
+    if (uncompressed == NULL)
+    {
+        this->flushing = true;
+        this->inputSame = zstError(ZSTD_endStream(this->context, &out));
```

Comment:
zstError should either return a boolean instead of size_t (seems only time the return value is assigned is to a boolean) or maybe have a conditional here like `zstError(ZSTD_endStream(this->context, &out)) != 0`

Reply:
Passing the size through is a lot more efficient in other places so I want to keep zstError as is. But I did update the syntax as suggested.